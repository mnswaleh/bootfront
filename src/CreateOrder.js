import React, { Component } from 'react';
import { connect } from "react-redux";
import { Form, Row, Col, Button } from 'react-bootstrap'
import { addOrder } from "./actions/index"

class Order extends Component {
    constructor() {
        super();

        this.state = {
            parcel: {
                "weight": 0,
                "pickup_location": "nyeri",
                "destination": "nyeri",
                "recipient": "",
                "parcel_details": ""
            }
        }
    }

    componentWillUnmount() {
        this._isMounted = false;
    }


    handleChange = (event) => {
        let form_data = this.state.parcel
        form_data[event.target.name] = event.target.value;
        this.setState({ parcel: form_data });
    }

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.addOrder(this.state.parcel);
    }

    render() {
        return (
            <div>
                <h2 className="text-center">Create Order</h2>

                <Form onSubmit={this.handleSubmit}>
                    <Form.Group as={Row} controlId="weight">
                        <Form.Label column sm={2}>Est Weight(Kg):</Form.Label>
                        <Col sm={10}>
                            <Form.Control type="number" name="weight" value={this.state.parcel['weight']} onChange={this.handleChange} />
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row} controlId="pickup_location">
                        <Form.Label column sm={2}>Pick Up Location</Form.Label>
                        <Col sm={10}>
                            <Form.Control as="select" name="pickup_location" value={this.state.parcel['pickup_location']} onChange={this.handleChange}>
                                <option value="nyeri">Nyeri</option>
                                <option value="kisumu">Kisumu</option>
                                <option value="mombasa">Mombasa</option>
                                <option value="kitale">Kitale</option>
                                <option value="eldoret">Eldoret</option>
                                <option value="naivasha">Naivasha</option>
                                <option value="kirinyaga">Kirinyaga</option>
                                <option value="mwea">Mwea</option>
                                <option value="narok">Narok</option>
                                <option value="kiambu">Kiambu</option>
                                <option value="nyahururu">Nyahururu</option>
                            </Form.Control>
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row} controlId="destination">
                        <Form.Label column sm={2}>Delivery Location</Form.Label>
                        <Col sm={10}>
                            <Form.Control as="select" name="destination" value={this.state.parcel['destination']} onChange={this.handleChange}>
                                <option value="nyeri">Nyeri</option>
                                <option value="kisumu">Kisumu</option>
                                <option value="mombasa">Mombasa</option>
                                <option value="kitale">Kitale</option>
                                <option value="eldoret">Eldoret</option>
                                <option value="naivasha">Naivasha</option>
                                <option value="kirinyaga">Kirinyaga</option>
                                <option value="mwea">Mwea</option>
                                <option value="narok">Narok</option>
                                <option value="kiambu">Kiambu</option>
                                <option value="nyahururu">Nyahururu</option>
                            </Form.Control>
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row} controlId="recipient">
                        <Form.Label column sm={2}>Receiver</Form.Label>
                        <Col sm={10}>
                            <Form.Control type="text" name="recipient" value={this.state.parcel['recipient']} onChange={this.handleChange} />
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row} controlId="parcel_details">
                        <Form.Label column sm={2}>Details</Form.Label>
                        <Col sm={10}>
                            <Form.Control as="textarea" rows="3" name="parcel_details" value={this.state.parcel['parcel_details']} onChange={this.handleChange} />
                        </Col>
                    </Form.Group>
                    <Form.Group as={Row}>
                        <Col sm={6}></Col>
                        <Col sm={6}>
                            <Button variant="primary" type="submit">Submit</Button>
                        </Col>
                    </Form.Group>
                </Form>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        addOrder: parcel => dispatch(addOrder(parcel))
    };
}

const mapStateToProps = state => {
    return { parcels: state.parcels };
};

const CreateOrder = connect(mapStateToProps, mapDispatchToProps)(Order);

export default CreateOrder;