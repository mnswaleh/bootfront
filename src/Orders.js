import React, { Component } from 'react';
import { connect } from "react-redux";
import { fetchParcels } from "./actions/index"


export class OrdersList extends Component {
  constructor() {
    super();
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  componentWillMount() {
    this.props.fetchParcels();
}

  render() {
    return (
      <div>
        <h2 className="text-center">Orders</h2>
        <table className="table">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">Order_No</th>
              <th scope="col">Destination</th>
              <th scope="col">Status</th>
            </tr>
          </thead>
          <tbody>
                {this.props.parcels.map((row, i) => (
                    <tr key={i}>
                        <th scope="row">{i + 1}</th>
                        <td>{row.parcel_id}</td>
                        <td>{row.destination}</td>
                        <td>{row.status}</td>
                    </tr>
                ))}
            </tbody>
        </table>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
      fetchParcels: parcels => dispatch(fetchParcels())
  };
}

const mapStateToProps = state => {
  return { parcels: state.parcels };
};

const Orders = connect(mapStateToProps, mapDispatchToProps)(OrdersList);
export default Orders;