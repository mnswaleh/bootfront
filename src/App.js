import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, NavLink } from 'react-router-dom';
import { Nav, NavItem } from 'react-bootstrap'
import Orders from './Orders';
import CreateOrder from './CreateOrder'

class App extends Component {
    render() {
        return (
            <Router>
                <div>
                    <Nav fill variant="pills" className="justify-content-center" activeKey="/" onSelect={selectedKey => alert(`selected ${selectedKey}`)}>
                        <NavItem><NavLink to={'/'}>Orders</NavLink></NavItem>
                        <NavItem><NavLink to={'/createorder'}> Create Order</NavLink></NavItem>
                        <NavItem><NavLink to={'/'}> Update Order</NavLink></NavItem>
                    </Nav>
                    <Switch>
                        <Route exact path='/' component={Orders}/>
                        <Route path='/createorder' component={CreateOrder} />
                    </Switch>
                </div>
            </Router>
        );
    }
}

export default App;