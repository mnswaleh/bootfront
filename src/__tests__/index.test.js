import React from 'react';
import { Provider } from "react-redux";
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallow } from 'enzyme';
import CreateOrder from '../CreateOrder';
import Orders from '../Orders';
import App from '../App';
import rootReducer from "../reducers/index";


let initialState = {
  parcels: [],
  parcel: {}
};

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
const store = mockStore(rootReducer)

describe('Reducers', () => {
  it('Should return initial state', () => {
    expect(rootReducer(undefined, {})).toEqual(initialState)
  });

  it('Should return parcels', () => {
    initialState.parcels = [{ "order_no": 3 }]
    expect(rootReducer(initialState.parcels, "FETCH_PARCEL")).toEqual(initialState.parcels)
  });

  it('Should return parcels', () => {
    initialState.parcel = { "order_no": 3 }
    expect(rootReducer(initialState.parcel, "ADD_PARCEL")).toEqual(initialState.parcel)
  });
});

describe('App', () => {
  it('Should render App', async (done) => {
    const app = await shallow(<Provider store={store}><App /></Provider>)
    expect(app).toMatchSnapshot();
    done()
  })
})

describe('Orders', () => {
  it('Should render Orders', async (done) => {
    const orders = await shallow(<Provider store={store}><Orders /></Provider>)
    expect(orders).toMatchSnapshot();
    done()
  })
})

describe('CreateOrder', () => {
  it('Should render create order form', async (done) => {
    const create_orders = await shallow(<Provider store={store}><CreateOrder /></Provider>)
    expect(create_orders).toMatchSnapshot();
    done()
  })
})