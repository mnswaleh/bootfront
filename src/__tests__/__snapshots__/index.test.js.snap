// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render App 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Array [],
    }
  }
>
  <App />
</ContextProvider>
`;

exports[`CreateOrder Should render create order form 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Array [],
    }
  }
>
  <Connect(Order) />
</ContextProvider>
`;

exports[`Orders Should render Orders 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Array [],
    }
  }
>
  <Connect(OrdersList) />
</ContextProvider>
`;
